app/collector.go

```diff
@@ -131,3 +136,56 @@ func (c *collector) clean() {
 	c.reports = cleanedReports
 	c.timestamps = cleanedTimestamps
 }
+
+// StaticCollector always returns the given report.
+type StaticCollector report.Report
+
+// Report returns a merged report over all added reports. It implements
+// Reporter.
+func (c StaticCollector) Report(context.Context) (report.Report, error) { return report.Report(c), nil }
+
+// Add adds a report to the collector's internal state. It implements Adder.
+func (c StaticCollector) Add(context.Context, report.Report) error { return nil }
+
+// WaitOn lets other conponents wait on a new report being received. It
+// implements Reporter.
+func (c StaticCollector) WaitOn(context.Context, chan struct{}) {}
+
+// UnWait lets other conponents stop waiting on a new report being received. It
+// implements Reporter.
+func (c StaticCollector) UnWait(context.Context, chan struct{}) {}
+
+// NewFileCollector reads and json parses the given path, returning a collector
```

Comment:
Delete "json", since it does more that that. Alternatively, update the docstring to say that it does json or msgpack based on the filename extension.